src/everything/resources/templates.ts

```diff
@@ -0,0 +1,211 @@
+ * @typedef {CompleteResourceTemplateCallback}
+ * @param {string} value - The input string value to be evaluated as a resource ID.
+ * @returns {string[]} Returns an array containing the input value if it represents a positive
+ * integer resource ID, otherwise returns an empty array.
+ */
+export const resourceIdForResourceTemplateCompleter: CompleteResourceTemplateCallback =
+  (value: string) => {
+    const resourceId = Number(value);
+
+    return Number.isInteger(resourceId) && resourceId > 0 ? [value] : [];
+  };
+
+const uriBase: string = "demo://resource/dynamic";
+const textUriBase: string = `${uriBase}/text`;
+const blobUriBase: string = `${uriBase}/blob`;
+const textUriTemplate: string = `${textUriBase}/{resourceId}`;
+const blobUriTemplate: string = `${blobUriBase}/{resourceId}`;
+
+/**
+ * Create a dynamic text resource
+ * - Exposed for use by embedded resource prompt example
+ * @param uri
+ * @param resourceId
+ */
+export const textResource = (uri: URL, resourceId: number) => {
+  const timestamp = new Date().toLocaleTimeString();
+  return {
+    uri: uri.toString(),
+    mimeType: "text/plain",
+    text: `Resource ${resourceId}: This is a plaintext resource created at ${timestamp}`,
+  };
+};
+
+/**
+ * Create a dynamic blob resource
+ * - Exposed for use by embedded resource prompt example
+ * @param uri
+ * @param resourceId
+ */
+export const blobResource = (uri: URL, resourceId: number) => {
+  const timestamp = new Date().toLocaleTimeString();
+  const resourceText = Buffer.from(
+    `Resource ${resourceId}: This is a base64 blob created at ${timestamp}`
+  ).toString("base64");
+  return {
+    uri: uri.toString(),
+    mimeType: "text/plain",
+    blob: resourceText,
+  };
+};
+
+/**
+ * Create a dynamic text resource URI
+ * - Exposed for use by embedded resource prompt example
+ * @param resourceId
+ */
+export const textResourceUri = (resourceId: number) =>
+  new URL(`${textUriBase}/${resourceId}`);
+
+/**
+ * Create a dynamic blob resource URI
+ * - Exposed for use by embedded resource prompt example
+ * @param resourceId
+ */
+export const blobResourceUri = (resourceId: number) =>
+  new URL(`${blobUriBase}/${resourceId}`);
+
+/**
+ * Parses the resource identifier from the provided URI and validates it
+ * against the given variables. Throws an error if the URI corresponds
+ * to an unknown resource or if the resource identifier is invalid.
+ *
+ * @param {URL} uri - The URI of the resource to be parsed.
+ * @param {Record<string, unknown>} variables - A record containing context-specific variables that include the resourceId.
+ * @returns {number} The parsed and validated resource identifier as an integer.
+ * @throws {Error} Throws an error if the URI matches unsupported base URIs or if the resourceId is invalid.
+ */
+const parseResourceId = (uri: URL, variables: Record<string, unknown>) => {
+  const uriError = `Unknown resource: ${uri.toString()}`;
+  if (
+    uri.toString().startsWith(textUriBase) &&
```

Comment:
```suggestion
    uri.toString().startsWith(textUriBase) ||
```

I think this is the intention?